File: frontend/src/types/session.ts
import type { PersistedChatMessage } from "@/types/chat";

export interface ChatSession {
  id: number;
  title: string;
  created_at: string;
  messages?: PersistedChatMessage[];
}

export interface CreateSessionResponse {
  session_id: number;
}

export interface RenameSessionRequest {
  title: string;
}

export interface RenameSessionResponse {
  id: number;
  title: string;
}

export interface DeleteSessionResponse {
  message: string;
}

File: backend/app/schemas/session.py
from datetime import datetime

from pydantic import BaseModel


class SessionResponse(BaseModel):

    id: int
    title: str
    created_at: datetime

    model_config = {
        "from_attributes": True,
    }


class CreateSessionResponse(BaseModel):

    session_id: int


class RenameSessionRequest(BaseModel):

    title: str


class RenameSessionResponse(BaseModel):

    id: int
    title: str

    model_config = {
        "from_attributes": True,
    }


class DeleteSessionResponse(BaseModel):

    message: str